internal/commands/build/build.go

```diff
@@ -128,37 +133,71 @@ func runBuild(dockerCli command.Cli, contextPath string, opt buildOptions) error
 	}
 	defer app.Cleanup()
 
-	bundle, err := buildImageUsingBuildx(app, contextPath, opt, dockerCli)
+	bndl, err := buildImageUsingBuildx(app, contextPath, opt, dockerCli)
 	if err != nil {
 		return err
 	}
 
-	var ref reference.Reference
-	ref, err = packager.GetNamedTagged(opt.tag)
+	out, err := getOutputFile(dockerCli.Out(), opt.quiet)
 	if err != nil {
 		return err
 	}
 
-	id, err := packager.PersistInBundleStore(ref, bundle)
+	id, err := persistTags(bndl, opt.tags, opt.imageIDFile, out, dockerCli.Err())
 	if err != nil {
 		return err
 	}
 
-	if opt.imageIDFile != "" {
-		if err = ioutil.WriteFile(opt.imageIDFile, []byte(id.Digest().String()), 0644); err != nil {
-			fmt.Fprintf(dockerCli.Err(), "Failed to write App image ID in %s: %s", opt.imageIDFile, err)
+	if opt.quiet {
+		_, err = fmt.Fprintln(dockerCli.Out(), id.Digest().String())
+		if err != nil {
+			return err
 		}
+		return nil
 	}
 
-	if opt.quiet {
-		fmt.Fprintln(dockerCli.Out(), id.Digest().String())
-		return err
+	return nil
+}
+
+func persistTags(bndl *bundle.Bundle, tags cliOpts.ListOpts, iidFile string, outWriter io.Writer, errWriter io.Writer) (reference.Digested, error) {
+	var id reference.Digested
+	var onceWriteIIDFile sync.Once
+	if tags.Len() == 0 {
```

Comment:
I think you can simplify a lot for this special case and remove the validation of empty tag.
What about this ?
```go
if tags.Len() == 0 {
    return persistInBundleStore(&onceWriteIIDFile, outWriter, errWriter, bndl, nil, iidFile)
}
for _, tag := rang tags.GetAll() {
    		ref, err := packager.GetNamedTagged(tag)
		if err != nil {
			return nil, err
		}
		id, err = persistInBundleStore(&onceWriteIIDFile, outWriter, errWriter, bndl, ref, iidFile)
		if err != nil {
			return nil, err
		}
		fmt.Fprintf(outWriter, "Successfully tagged app %s\n", ref.String())
}
```

Reply:
Cool! Actually that was my first approach. But I didn't want to call `persistInBundleStore` twice, so I had shared feelings about it. But you convinced me